### PR TITLE
add attach command for crictl

### PR DESCRIPTION
**Pull request by heartlock**
Signed-off-by: heartlock <21521209@zju.edu.cn>

Closes: #38 
@feiskyer @xlgao-zju

### CODE REVIEW

**Comment by feiskyer** [comment]
> ref: #38

Please use `Closes: #38` or `Fixes #38`, which could close the issue automatically when this PR is merged.

File: cmd/crictl/attach.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	Usage: "attach a running container",
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:  "id",
+			Value: "",
+			Usage: "id of the container",
+		},
+		cli.BoolFlag{
+			Name:  "tty",
+			Usage: "attach a peocess in a tty",
```
Comment (feiskyer) [inline cmd/crictl/attach.go]: s/attach a peocess in a tty/Stdin is a TTY

File: cmd/crictl/attach.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+		},
+		cli.BoolFlag{
+			Name:  "stdin",
+			Usage: "stream stdin",
```
Comment (feiskyer) [inline cmd/crictl/attach.go]: s/stream stdin/Pass stdin to the container

File: cmd/crictl/attach.go (hunk 1 above)
Comment (feiskyer) [inline cmd/crictl/attach.go]: Also add a short '-t'

File: cmd/crictl/attach.go (hunk 2 above)
Comment (feiskyer) [inline cmd/crictl/attach.go]: also add a short '-i', e.g. `stdin, i`

File: cmd/crictl/util.go [hunk 3]
```diff
@@ -56,6 +56,14 @@ type execOptions struct {
 	// Command to exec
 	cmd []string
 }
+type attachOptions struct {
+	// id of container
+	id string
+	// Whether the process being attached is running in a tty
```
Comment (feiskyer) [inline cmd/crictl/util.go]: Whether the stdin is TTY

File: cmd/crictl/util.go [hunk 4]
```diff
@@ -56,6 +56,14 @@ type execOptions struct {
...
+	tty bool
+	// Whether to stream stdin
```
Comment (feiskyer) [inline cmd/crictl/util.go]: Whether pass Stdin to container

**Comment by heartlock** [comment]
@feiskyer  updated

**Pull request merged**